src/periphery/GlpVaultHarvestor.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "BoringSolidity/interfaces/IERC20.sol";
+import "BoringSolidity/BoringOwnable.sol";
+import "OpenZeppelin/utils/Address.sol";
+import "interfaces/IGmxGlpRewardHandler.sol";
+import "interfaces/IMimCauldronDistributor.sol";
+import "interfaces/IGmxRewardRouterV2.sol";
+import "interfaces/IGmxRewardTracker.sol";
+
+/// @dev Glp harvester version that swap the reward to USDC to mint glp
+/// and transfer them back in GmxGlpVault token for auto compounding
+contract GlpVaultHarvestor is BoringOwnable {
+    using Address for address;
+
+    event OperatorChanged(address indexed, bool);
+    event RewardTokenChanged(IERC20 indexed, IERC20 indexed);
+    event OutputTokenChanged(IERC20 indexed, IERC20 indexed);
+    event RewardRouterV2Changed(IGmxRewardRouterV2 indexed, IGmxRewardRouterV2 indexed);
+    error NotAllowedOperator();
+    error ReturnRewardBalance(uint256 balance);
+
+    IGmxGlpRewardHandler public immutable wrapper;
+
+    IERC20 public rewardToken;
+    IERC20 public outputToken;
+    IGmxRewardRouterV2 public rewardRouterV2;
```

Comment:
there is the need to track both old and new rewardRouter,
consider calling one mintRewardRouter and claimRewardRouter, the former is the new one the latter the old one